packages/context/test/acceptance/context-view.acceptance.ts

```diff
@@ -0,0 +1,182 @@
+import {expect} from '@loopback/testlab';
+import {
+  Context,
+  ContextView,
+  ContextEventListener,
+  Getter,
+  inject,
+  ContextEventType,
+  Binding,
+} from '../..';
+
+describe('ContextView - watches matching bindings', () => {
+  let server: Context;
+  let contextWatcher: ContextView;
+  beforeEach(givenControllerWatcher);
+
+  it('watches matching bindings', async () => {
+    // We have ctx: 1, parent: 2
+    expect(await getControllers()).to.eql(['1', '2']);
+    server.unbind('controllers.1');
+    // Now we have parent: 2
+    expect(await getControllers()).to.eql(['2']);
+    server.parent!.unbind('controllers.2');
+    // All controllers are gone from the context chain
+    expect(await getControllers()).to.eql([]);
+    // Add a new controller - ctx: 3
+    givenController(server, '3');
+    expect(await getControllers()).to.eql(['3']);
+  });
+
+  function givenControllerWatcher() {
+    server = givenServerWithinAnApp();
+    contextWatcher = server.watch(Context.bindingTagFilter('controller'));
+    givenController(server, '1');
+    givenController(server.parent!, '2');
+  }
+
+  function givenController(_ctx: Context, _name: string) {
+    class MyController {
+      name = _name;
+    }
+    _ctx
+      .bind(`controllers.${_name}`)
+      .toClass(MyController)
+      .tag('controller');
+  }
+
+  async function getControllers() {
+    // tslint:disable-next-line:no-any
+    return (await contextWatcher.values()).map((v: any) => v.name);
+  }
+});
+
+describe('@inject.filter - injects a live collection of matching bindings', async () => {
```

Comment:
The name passed to `describe` is usually supposed to describe what is being tested (the subject under test), not the actual behavior/test scenarios.

Also `@inject.filter` was renamed to `@inject.view` IIUC.

Last but not least, test suites should not do any async computation, i.e. should be defined as a regular function (not an async function).

My proposal:

```suggestion
describe('@inject.view', () => {
```